src/compressed_tensors/quantization/lifecycle/initialize.py

```diff
@@ -249,32 +248,31 @@ def initialize_qparams(
         assert False, f"Unknown strategy {strategy}"
 
     # 2. Identify quantization scale and zp dtype
-    scale_dtype = observed_dtype
-
-    if is_fp4(quantization_args=quantization_args):
-        scale_dtype = zp_dtype = FP8_E4M3_DATA.dtype
-    else:
-        # TODO: consider erroring out in the future as if the dtype if not one of these,
-        # there is likely bug
-        if scale_dtype not in [
+    if quantization_args.scale_dtype is None:
+        if observed_dtype not in [
             torch.float16,
             torch.bfloat16,
             torch.float32,
             torch.float64,
         ]:
-            scale_dtype = torch.bfloat16
-        zp_dtype = quantization_args.pytorch_dtype()
+            observed_dtype = torch.float16
+        quantization_args.scale_dtype = observed_dtype
```

Comment:
I don't like waiting this long to modify qargs (for fear that something might read it before this override happens. Can we write a validator for `scale_dtype` in `QuantizationArgs`?